libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m

```diff
@@ -76,9 +76,7 @@ - (void)setEnqueuedNetwork:(CSFNetwork *) network {
 		_enqueuedNetwork = network;
         // add observers to the new network
         if (_enqueuedNetwork) {
-            if (!self.baseURL.scheme && !self.baseURL.host) {
-                // only set base URL to apiURL if baseURL is not already specified as absolute URL with it's own host
-                // this check is necessary as there are salesforce URL that is content server based and not API based
+            if ([self shouldUpdateBaseUrl]) {
```

Comment:
consolidate the logic in a method as it will be used below again in the key value observer